### ISSUE TITLE
Confusing error message on definition type mismatch

### ISSUE THREAD

**Issue by gabrielhdt** [Deducteam/lambdapi#522]
The following file
```
symbol A: TYPE
symbol B: TYPE
symbol a: A
symbol b: B ≔ a
```

returns
> `[confused.lp, 4:15-16] Not a typing goal`

when calling
```
lambdapi check confused.lp
```

on the terminal, which is not very informative.

As far as I understand it, it is a result of calling `refine` on the definition `a` in
https://github.com/Deducteam/lambdapi/blob/038bc694d45d5069cf3488682928d2315f0b0d17/src/core/handle.ml#L430
which in turns triggers
https://github.com/Deducteam/lambdapi/blob/038bc694d45d5069cf3488682928d2315f0b0d17/src/core/tactics.ml#L42

**Comment by fblanqui** [comment]
It should be fixed in #523. Thanks to try this branch to see what is the output now. If you are happy, I will merge it.

**Comment by gabrielhdt** [comment]
Yes I have tested (but forgotten to report), and it's way better, thank you!
